src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -58,7 +62,19 @@ internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T> : ArrayPool
         [ThreadStatic]
         private static Bucket[] t_tlsBuckets;
 
-        private static ConditionalWeakTable<Bucket[], object> s_allTlsBuckets = new ConditionalWeakTable<Bucket[], object>();
+        private int _callbackCreated;
+
+        /// <summary>
+        /// Used to keep track of all thread local buckets for trimming if needed
+        /// </summary>
+        private static readonly ConditionalWeakTable<Bucket[], object> s_AllTlsBuckets;
+
+        private unsafe static int s_typeSize = Unsafe.SizeOf<T>();
```

Comment:
`Unsafe.SizeOf` always turns in JIT time constant. Why cache it?

Reply:
Because I didn't realize that. :)